feat: Add templates for xunit.v3

linkdotnet opened the pull request:
Added another template for `xunit.v3`.

egil commented:
Should we upgrade dependencies on the other templates too?

egil commented:
Is there any conflict between bUnits TestContext and Xunits? Haven't tested yet myself.

linkdotnet commented:
> Is there any conflict between bUnits TestContext and Xunits? Haven't tested yet myself.

My understanding is that you either use the static object aka `TestContext.Current` or via DI `ITestContextAccessor` - so we should be safe.

linkdotnet commented:
> Should we upgrade dependencies on the other templates too?

Sure why not

linkdotnet commented:
I did the following:
 * Added the xunit.v3 template
 * Updated dependencies
 * Removed `net6.0` and `net7.0` from our target framework choices as they are EOL

linkdotnet commented:
> Is there any conflict between bUnits TestContext and Xunits? Haven't tested yet myself.

You were right - there are conflicts.

egil reviewed (approved):
Excellent

egil commented:
> > Is there any conflict between bUnits TestContext and Xunits? Haven't tested yet myself.
> 
> 
> 
> You were right - there are conflicts.

Than we can change our default import to be:

`using BunitContext = Bunit.TestContext;`

linkdotnet commented:
> using BunitContext = Bunit.TestContext;

You mean for that specific template? It might be confusing and less verbose (especially regarding all our docs).
Therefore I opted out of this behavior.

linkdotnet commented:
@egil Can we merge this or do you want to rename `Bunit.TestContext` to `BunitTestContext` in the template?
See last comment from my side

egil commented:
> @egil Can we merge this or do you want to rename `Bunit.TestContext` to `BunitTestContext` in the template?
> 
> See last comment from my side

An important point of having a template is that works out of the box for the users.

For now, I guess we can just have test class examples inherit from the fqns, e.g. `public class CounterTests : Bunit.TeatContext`

linkdotnet commented:
> > @egil Can we merge this or do you want to rename `Bunit.TestContext` to `BunitTestContext` in the template?
> > See last comment from my side
> 
> An important point of having a template is that works out of the box for the users.
> 
> For now, I guess we can just have test class examples inherit from the fqns, e.g. `public class CounterTests : Bunit.TeatContext`

I absolutely agree - and if it wouldn't be for `v2`, I also would go down the route of using a global alias (even though it hides the fact and might be confusing as well).
